### ISSUE TITLE
Incorrect padding for keras mode="same" in dilated convolution

### ISSUE THREAD

**Issue by thefiddler** [dmlc/nnvm#307]
When converting keras models using `mode="same"` in convolutions and pooling, the [calculated padding shape](https://github.com/dmlc/nnvm/blob/master/python/nnvm/frontend/keras.py#L185-L197) is different between keras and nnvm. The attached program highlights the issue.

```bash
$ python3 test_padding.py
Output shape c1_same (None, 128, 128, 1)
Output shape c1_valid (None, 123, 122, 1)
Output shape c2_same (None, 128, 128, 1)
Output shape c2_valid (None, 118, 116, 1)
Output shape mp_same (None, 128, 128, 1)
Output shape mp_valid (None, 123, 122, 1)
Keras padding shape c1_same:  (5, 6, 0)
Keras padding shape c2_same:  (10, 12, 0)
Keras padding shape mp_same:  (5, 6, 0)
NNVM padding shape c1_same: (5, 6, 0)
NNVM padding shape c2_same: (5, 6, 0)
NNVM padding shape mp_same: (5, 6, 0)
```

Note that you would need to add a print statement to python/nnvm/frontend/keras.py on lines 150 and 258 (or use a debugger) to observe the calculated values.

test_padding.py:
```python
import nnvm
from keras.models import Model
from keras.layers import *
from keras.layers.pooling import MaxPooling2D
from keras.optimizers import *
from keras.activations import linear

input = Input((128, 128, 1))
c1_same  = Convolution2D(1, (6, 7), dilation_rate=(1, 1), padding="same")(input)
c1_valid  = Convolution2D(1, (6, 7), dilation_rate=(1, 1), padding="valid")(input)
c2_same = Convolution2D(1, (6, 7), dilation_rate=(2, 2), padding="same")(input)
c2_valid = Convolution2D(1, (6, 7), dilation_rate=(2, 2), padding="valid")(input)
mp_same  = MaxPooling2D((6, 7), strides=(1, 1), padding="same")(input)
mp_valid  = MaxPooling2D((6, 7), strides=(1, 1), padding="valid")(input)
model = Model([input], [c1_same, c1_valid, c2_same, c2_valid, mp_same, mp_valid])

print("Output shape c1_same", model.output_shape[0])
print("Output shape c1_valid", model.output_shape[1])
print("Output shape c2_same", model.output_shape[2])
print("Output shape c2_valid", model.output_shape[3])
print("Output shape mp_same", model.output_shape[4])
print("Output shape mp_valid", model.output_shape[5])

# calculate the applied paddings on each axis
# skip the batchsize dimension, as that is normally None at this point

padding = tuple((s - t for s, t in zip(model.output_shape[0][1:], model.output_shape[1][1:])))
print("Keras padding shape c1_same: ", padding)

padding = tuple((s - t for s, t in zip(model.output_shape[0][1:], model.output_shape[3][1:])))
print("Keras padding shape c2_same: ", padding)

padding = tuple((s - t for s, t in zip(model.output_shape[0][1:], model.output_shape[5][1:])))
print("Keras padding shape mp_same: ", padding)

_ = nnvm.frontend.from_keras(Model([input], [c1_same]))
_ = nnvm.frontend.from_keras(Model([input], [c2_same]))
_ = nnvm.frontend.from_keras(Model([input], [mp_same]))
```

**Comment by thefiddler** [comment]
This is fixed by #322
